runtime/interp.c

```diff
@@ -687,23 +690,25 @@ value caml_interprete(code_t prog, asize_t prog_size)
       *--sp = accu;
       /* Fallthrough */
     Instruct(GETGLOBAL):
-      accu = Field(caml_read_root(caml_global_data), *pc);
+      accu = Field(caml_global_data, *pc);
       pc++;
       Next;
 
     Instruct(PUSHGETGLOBALFIELD):
       *--sp = accu;
       /* Fallthrough */
     Instruct(GETGLOBALFIELD): {
-      accu = Field(caml_read_root(caml_global_data), *pc);
+      accu = Field(caml_global_data, *pc);
       pc++;
       accu = Field(accu, *pc);
       pc++;
       Next;
     }
 
     Instruct(SETGLOBAL):
-      caml_modify(&Field(caml_read_root(caml_global_data), *pc), accu);
+      global_data = caml_global_data;
```

Comment:
This look "interesting". Perhaps a comment would be useful. Is this the only place where `global_data` is used?

Reply:
Yeah, this is the only place `global_data` is used. It's just an intermediate location to store `caml_global_data` and call `caml_modify`, since we can't use `caml_modify` directly on `caml_global_data` and will need to use `caml_modify_generational_global_root` to update it. Will add a comment conveying this.